lighthouse-core/computed/resource-summary.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const makeComputedArtifact = require('./computed-artifact.js');
+const NetworkRecords = require('./network-records.js');
+const MainResource = require('./main-resource.js');
+const URL = require('../lib/url-shim.js');
+
+class ResourceSummary {
+  /**
+   * @param {LH.Artifacts.NetworkRequest} record
+   * @return {LH.Budget.ResourceType}
+   */
+  static determineResourceType(record) {
+    /** @type {Partial<Record<LH.Crdp.Page.ResourceType, LH.Budget.ResourceType>>} */
+    const mapping = {
+      'Stylesheet': 'stylesheet',
+      'Image': 'image',
+      'Media': 'media',
+      'Font': 'font',
+      'Script': 'script',
+      'Document': 'document',
+    };
+    if (!record.resourceType) return 'other';
+    return mapping[record.resourceType] || 'other';
+  }
+
+  /**
+   * @param {Array<LH.Artifacts.NetworkRequest>} networkRecords
+   * @param {string} mainResourceURL
+   * @return {Object<string,{resourceType: string, count: number, size: number}>}
+   */
+  static summarize(networkRecords, mainResourceURL) {
+    /** @type {Record<LH.Budget.ResourceType,{resourceType: LH.Budget.ResourceType, count: number, size: number}>} */
+    const map = {
+      'stylesheet': {count: 0, size: 0, resourceType: 'stylesheet'},
+      'image': {count: 0, size: 0, resourceType: 'image'},
+      'media': {count: 0, size: 0, resourceType: 'media'},
+      'font': {count: 0, size: 0, resourceType: 'font'},
+      'script': {count: 0, size: 0, resourceType: 'script'},
+      'document': {count: 0, size: 0, resourceType: 'document'},
+      'other': {count: 0, size: 0, resourceType: 'other'},
+      'total': {count: 0, size: 0, resourceType: 'total'},
+      'third-party': {count: 0, size: 0, resourceType: 'third-party'},
+    };
+
+    for (const record of networkRecords) {
+      const type = this.determineResourceType(record);
+      map[type].count++;
+      map[type].size = map[type].size + record.transferSize;
+
+      map.total.count++;
+      map.total.size = map.total.size + record.transferSize;
+
+      if (!URL.rootDomainsMatch(record.url, mainResourceURL)) {
```

Comment:
ah. i had always thought the non-3P items were implicitly 1P. 

but you're thinking they're straight up totals... 
naming-wise that makes sense, i suppose.

perhaps in the future we add support for budgeting 1P resource type totals?